diff --git a/bindings/python/proxsuite/__init__.py b/bindings/python/proxsuite/__init__.py
index 01ae2109..e7b49193 100644
--- a/bindings/python/proxsuite/__init__.py
+++ b/bindings/python/proxsuite/__init__.py
@@ -1,6 +1,10 @@
 import platform

-if "arm" not in platform.processor():
+machine = platform.machine()
+has_vectorization_instructions = not machine.startswith(
+    ("arm", "aarch64", "power", "ppc64", "s390x", "sparc")
+)
+if has_vectorization_instructions:
     from . import instructionset


@@ -16,7 +20,7 @@ def load_main_module(globals):
         except ModuleNotFoundError:
             return False

-    if "arm" not in platform.processor():
+    if has_vectorization_instructions:
         all_modules = [
             ("proxsuite_pywrap_avx512", instructionset.has_AVX512F),
             ("proxsuite_pywrap_avx2", instructionset.has_AVX2),
@@ -32,3 +36,5 @@ def load_main_module(globals):
 load_main_module(globals=globals())
 del load_main_module
 del platform
+del has_vectorization_instructions
+del machine
diff --git a/include/proxsuite/linalg/dense/core.hpp b/include/proxsuite/linalg/dense/core.hpp
index 72c0b31b..89d3511e 100644
--- a/include/proxsuite/linalg/dense/core.hpp
+++ b/include/proxsuite/linalg/dense/core.hpp
@@ -9,7 +9,8 @@
 #include <proxsuite/linalg/veg/util/assert.hpp>
 #include <proxsuite/linalg/veg/memory/dynamic_stack.hpp>

-#ifndef __aarch64__
+#if !(defined(__aarch64__) || defined(__PPC64__) || defined(__ppc64__) ||      \
+      defined(_ARCH_PPC64))
 #include <immintrin.h>
 #endif


